Formula/pynvim.rb

```diff
@@ -0,0 +1,24 @@
+class Pynvim < Formula
+  include Language::Python::Virtualenv
+
+  desc "Python client and plugin host for Nvim"
+  homepage "http://pynvim.readthedocs.io/en/latest/"
+  url "https://files.pythonhosted.org/packages/7a/01/2d0898ba6cefbe2736283ee3155cba1c602de641ca5667ac55a0e4857276/pynvim-0.4.3.tar.gz"
+  sha256 "3a795378bde5e8092fbeb3a1a99be9c613d2685542f1db0e5c6fd467eed56dff"
+  license "Apache-2.0"
+
+  depends_on "python@3.9"
+
+  resource "wcwidth" do
+    url "https://files.pythonhosted.org/packages/59/04/87fc6708659c2ed3b0b6d4954f270b6e931def707b227c4554f99bd5401e/msgpack-1.0.2.tar.gz"
+    sha256 "fae04496f5bc150eefad4e9571d1a76c55d021325dcd484ce45065ebbdd00984"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system "false"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I updated the test section according to your suggestions and with reference to [NumpyAT116-Forumla](https://github.com/Homebrew/homebrew-core/blob/master/Formula/numpy%401.16.rb)

Reply:
Sorry @chenrui333  I messed up with the rebase, is there a change to remove the other 20 files from PR?